web/src/app/users/UserDetails.js

```diff
@@ -53,128 +57,126 @@ function serviceCount(onCallSteps = []) {
   return Object.keys(svcs).length
 }
 
-@withStyles(styles)
-export default class UserDetails extends React.PureComponent {
-  static propTypes = {
-    userID: p.string.isRequired,
-    readOnly: p.bool,
-  }
+export default function UserDetails(props) {
+  const classes = useStyles()
 
-  state = {
-    createCM: false,
-    createNR: false,
-    showVerifyDialogByID: null,
-  }
+  const [disclaimer] = useConfigValue('General.NotificationDisclaimer')
+  const [createCM, setCreateCM] = useState(false)
+  const [createNR, setCreateNR] = useState(false)
+  const [showVerifyDialogByID, setShowVerifyDialogByID] = useState(null)
 
-  render() {
-    return (
-      <Query
-        query={query}
-        variables={{ id: this.props.userID }}
-        render={({ data }) => this.renderData(data.user)}
-      />
-    )
+  const { data, loading, error } = useQuery(query, {
+    variables: { id: props.userID },
+  })
+
+  if (error) return <GenericError error={error.message} />
+  if (!_.get(data, 'user.id')) {
```

Comment:
Can we use a ternary here instead? That avoids nesting `if` statements and makes it a bit more concise.

```suggestion
  if (!_.get(data, 'user.id')) return loading ? <Spinner /> : <ObjectNotFound />
```